python/lsst/pipe/tasks/dcrAssembleCoadd.py

```diff
@@ -0,0 +1,980 @@
+        bbox : lsst.afw.geom.box.Box2I
+            Sub-region to coadd
+        tempExpRefList : List of ButlerDataRefs
+            The data references to the input warped exposures.
+        imageScalerList : List of image scalers
+            The image scalars correct for the zero point of the exposures.
+        weightList : list of floats
+            The weight to give each input exposure in the coadd
+        spanSetMaskList : List of Dicts containing spanSet lists, or None
+            Each element is dict with keys = mask plane name to add the spans to
+        statsFlags : lsst.afw.math.Property
+            Statistics settings for coaddition.
+        statsCtrl : lsst.afw.math.StatisticsControl
+            Statistics control object for coadd
+
+        Returns
+        -------
+        `float`
+            Quality of fit metric for all input exposures, within the sub-region
+        """
+        maskMap = self.setRejectedMaskMapping(statsCtrl)
+        clipped = afwImage.Mask.getPlaneBitMask("CLIPPED")
+        tempExpName = self.getTempExpDatasetName(self.warpType)
+        modelWeightList = [1.0]*self.config.dcrNumSubfilters
+        dcrModelCut = [model[bbox, afwImage.PARENT] for model in dcrModels]
+        modelSum = afwMath.statisticsStack(dcrModelCut, statsFlags, statsCtrl, modelWeightList,
+                                           clipped,
+                                           maskMap)
+        significanceImage = np.abs(modelSum.image.array)
+        weight = 0
+        metric = 0.
+        metricList = {}
+        zipIterables = zip(tempExpRefList, weightList, imageScalerList, spanSetMaskList)
+        for tempExpRef, expWeight, imageScaler, altMaskSpans in zipIterables:
+            exposure = tempExpRef.get(tempExpName + "_sub", bbox=bbox)
+            imageScaler.scaleMaskedImage(exposure.maskedImage)
+            singleMetric = self.calculateSingleConvergence(exposure, dcrModels, significanceImage, statsCtrl,
+                                                           altMaskSpans=altMaskSpans)
+            metric += singleMetric*expWeight
+            metricList[tempExpRef.dataId["visit"]] = singleMetric
+            weight += expWeight
+        self.log.info("Individual metrics:\n%s", metricList)
+        if weight == 0:
+            return 1.
+        else:
+            return metric/weight
+
+    def calculateSingleConvergence(self, exposure, dcrModels, significanceImage, statsCtrl,
+                                   altMaskSpans=None):
+        """Calculate a quality of fit metric for a single matched template.
+
+        Parameters
+        ----------
+        exposure : lsst.afw.image.exposure.exposure.ExposureF
+            The input warped exposure to evaluate.
+        dcrModels : list of lsst.afw.image.maskedImageF
+            A list of masked images, each containing the model for one subfilter
+        significanceImage : `np.ndarray`
```

Comment:
I don't think the docs know anything about our namespace renaming, so this should be `numpy.ndarray`.